pkg/cmd/step/verify/step_verify_preinstall.go

```diff
@@ -558,18 +559,44 @@ func (o *StepVerifyPreInstallOptions) gatherRequirements(requirements *config.Re
 		requirements.Cluster.GitKind = "gitlab"
 	}
 
-	err = requirements.SaveConfig(requirementsFileName)
-	if err != nil {
-		return nil, errors.Wrap(err, "error saving requirements file")
-	}
+	var err error
+	if requirements.Cluster.EnvironmentGitOwner == "" {
+		requirements.Cluster.EnvironmentGitOwner, err = util.PickValue(
+			"Git Owner name for environment repositories",
+			"",
+			true,
+			"Jenkins X leverages GitOps to track and control what gets deployed into environments.  "+
+				"This requires a Git repository per environment. "+
+				"This question is asking for the Git Owner where these repositories will live.",
+			o.In, o.Out, o.Err)
+		if err != nil {
+			return errors.Wrap(err, "error configuring git owner for env repositories")
+		}
 
-	if requirements.Cluster.EnvironmentGitPublic {
-		log.Logger().Infof("Will create %s environment repos, if you want to create %s environment repos, please set %s to %s jx-requirements.yaml", util.ColorInfo("public"), util.ColorInfo("private"), util.ColorInfo("environmentGitPublic"), util.ColorInfo("false"))
-	} else {
-		log.Logger().Infof("Will create %s environment repos, if you want to create %s environment repos, please set %s to %s in jx-requirements.yaml", util.ColorInfo("private"), util.ColorInfo("public"), util.ColorInfo("environmentGitPublic"), util.ColorInfo("true"))
+		if requirements.Cluster.EnvironmentGitPublic {
+			log.Logger().Infof("Environment repos will be %s, if you want to create %s environment repos, please set %s to %s jx-requirements.yaml", util.ColorInfo("public"), util.ColorInfo("private"), util.ColorInfo("environmentGitPublic"), util.ColorInfo("false"))
+		} else {
+			err = o.verifyPrivateRepos(requirements)
+			if err != nil {
+				return err
+			}
+		}
 	}
+	return nil
+}
 
-	return requirements, nil
+func (o *StepVerifyPreInstallOptions) verifyPrivateRepos(requirements *config.RequirementsConfig) error {
+	log.Logger().Infof("Environment repos will be %s, if you want to create %s environment repos, please set %s to %s in jx-requirements.yaml", util.ColorInfo("private"), util.ColorInfo("public"), util.ColorInfo("environmentGitPublic"), util.ColorInfo("true"))
+
+	if requirements.Cluster.GitKind == "github" {
+		message := fmt.Sprintf("If '%s' is an GitHub organisation it needs to have a paid subscription to create private repos. Do you wish to continue?", requirements.Cluster.EnvironmentGitOwner)
+		help := fmt.Sprint("GitHub organisation on a free plan cannot create private repositories. You either need to upgrade, use a GitHub user instead or use public repositories.")
+		confirmed := util.Confirm(message, false, help, o.In, o.Out, o.Err)
```

Comment:
So here we have the additional confirmation we talked about for boot. Do we need something for `jx install` as well?

@jstrachan, not sure how I would do something special for batch mode here? Something needs to be set in the requirements file if in batch mode.

Reply:
I think this will prompt now always when `EnvironmentGitPublic` is set `false` which is a problem when boot is executed in the pipeline. You can use the `InCluster()` to detect when this is running in a pipeline and skip the `confirm`.

Reply:
I think if o.BatchMode might be even a better condition to check. If batch mode is enabled, don't confirm, wdyt?